electrum/gui/qml/qeqrscanner.py

```diff
@@ -49,34 +57,34 @@ def open(self):
         if 'ANDROID_DATA' not in os.environ:
             self._scan_qr_non_android()
             return
-        SimpleScannerActivity = autoclass("org.electrum.qr.SimpleScannerActivity")
-        intent = jIntent(jpythonActivity, SimpleScannerActivity)
+        jSimpleScannerActivity = autoclass("org.electrum.qr.SimpleScannerActivity")
+        intent = jIntent(jpythonActivity, jSimpleScannerActivity)
         intent.putExtra(jIntent.EXTRA_TEXT, jString(self._hint))
 
-        def on_qr_result(requestCode, resultCode, intent):
-            try:
-                if resultCode == -1:  # RESULT_OK:
-                    #  this doesn't work due to some bug in jnius:
-                    # contents = intent.getStringExtra("text")
-                    contents = intent.getStringExtra(jString("text"))
-                    #self._logger.info(f"on_qr_result. {contents=!r}")
-                    self.scanData = contents
-                    self.found.emit()
-            except Exception as e:  # exc would otherwise get lost
-                send_exception_to_crash_reporter(e)
-            finally:
-                activity.unbind(on_activity_result=on_qr_result)
-        activity.bind(on_activity_result=on_qr_result)
+        activity.bind(on_activity_result=self.on_qr_activity_result)
         jpythonActivity.startActivityForResult(intent, 0)
 
+    def on_qr_activity_result(self, requestCode, resultCode, intent):
+        try:
+            if resultCode == -1:  # RESULT_OK:
+                contents = intent.getStringExtra(jString("text"))
+                self.scanData = contents
+                self.found.emit()
+        except Exception as e:  # exc would otherwise get lost
+            send_exception_to_crash_reporter(e)
+        finally:
+            self.finished.emit()
+
     @pyqtSlot()
-    def close(self):
-        pass
+    def _unbind(self):
+        if 'ANDROID_DATA' in os.environ:
+            activity.unbind(on_activity_result=self.on_qr_activity_result)
```

Comment:
Is it ok for this to run twice?
`finished` and `destroyed` might both trigger it.

Reply:
https://github.com/kivy/python-for-android/blob/5dc17e16217620daf76d58ac6d20e4f9a9d4ff8b/pythonforandroid/recipes/android/src/android/activity.py#L57-L59
Looks like it is not thread-safe but otherwise fine to call it multiple times.

Reply:
yes, that's ok, see https://github.com/kivy/python-for-android/blob/5dc17e16217620daf76d58ac6d20e4f9a9d4ff8b/pythonforandroid/recipes/android/src/android/activity.py#L52-L63

Reply:
> Looks like it is not thread-safe but otherwise fine to call it multiple times.

The java counterpart is 'thread-safe' by using a `synchronized` block, although judging from the exception it isn't really blocking on the lock (maybe due to python-java boundary crossing?)

Reply:
We could add our own locking via threading.Lock() around `activity.unbind`. Though maybe the pyqt signalling already makes it indirectly invoked from the safe thread, I'm not sure. Anyway, up to you, if you think it's worth it. Otherwise the diff looks good.

Reply:
the Qt signal handlers are called in-sequence, from the UI thread. I think we're good here. 
If we do want to add a lock, I think the proper place is in P4A